nncf/errors.py

```diff
@@ -0,0 +1,111 @@
+class ValidationError(Exception):
+    """
+        Raised when an internal NNCF validation check fails, for example, if the user supplied an invalid or inconsistent set of arguments.
+    """
+    pass
+
+class InternalError(Exception):
+    """
+    Raised when an internal error occurs within the NNCF framework.
+
+    This exception is raised when an unexpected internal error occurs during the execution
+    of NNCF. It indicates a situation where the code encountered an unexpected condition.
+
+    """
+    pass
+
+class UnsupportedDatasetError(Exception):
+    """
+    Raised when an unsupported dataset is encountered
+    """
+    pass
+
+class InvalidFolderPathError(Exception):
+    """
+    Raised when an invalid folder path is provided
+    """
+    pass
+
+class UnsupportedBackendError(Exception):
+    """
+    Raised when an unsupported backend is specified
+    """
+    pass
+
+class InconsistentRegistryError(Exception):
+    """
+    Raised when there is an inconsistency in the registry
+    """
+    pass
+
+class InvalidQuantizerGroupError(Exception):
+    """
+    Raised when an invalid quantizer group is encountered.
+    """
+    pass
+
+class InstallationError(Exception):
+    """
+    Raised when an error occurs during installation.
+    """
+    pass
+
+class UnsupportedModelError(Exception):
+    """
+    Raised when an unsupported model is encountered.
+    """
+    pass
+
+class UnsupportedVersionError(Exception):
+    """
+    Raised when an unsupported version is encountered.
+    """
+    pass
+
+class ModuleNotFoundError(Exception):
+    """
+    Raised when a required module is not found.
+    """
+    pass
+
+class ParameterNotFoundError(Exception):
```

Comment:
This `Parameter` is semantically `torch.nn.Parameter`, while the `Parameter` in the `ParameterNotSupportedError` below is a generic "compression algorithm API parameter", which may be confusing. Based on the usage, I propose to merge this `ParameterNotFoundError` into `nncf.InternalError`. Also, for the remaining `nncf.ParameterNotSupportedError` please add some words into the docstring explaining that the `Parameter` here actually refers to a compression algo parameter (such as a parameter passed in an argument of an API call).